x-pack/legacy/plugins/ml/server/models/calendar/calendar_manager.ts

```diff
@@ -6,17 +6,46 @@
 
 import _ from 'lodash';
 import Boom from 'boom';
-import { EventManager } from './event_manager';
+import { EventManager, CalendarEvent } from './event_manager';
+
+interface BasicCalendar {
+  job_ids: string[];
+  description?: string;
+  events: CalendarEvent[];
+}
+
+export interface Calendar extends BasicCalendar {
+  calendar_id: string;
+}
+
+export interface FormCalendar extends BasicCalendar {
+  calendarId: string;
+}
 
 export class CalendarManager {
-  constructor(callWithRequest) {
-    this.callWithRequest = callWithRequest;
-    this.eventManager = new EventManager(callWithRequest);
+  private isLegacy: boolean;
+  private client: any;
+  private eventManager: any;
+
+  constructor(isLegacy: boolean, client: any) {
+    this.isLegacy = isLegacy;
+    this.client = client;
```

Comment:
it might be worth making the `isLegacy` check once here in the constructor and setting `this.client` as either `client` or `client.ml!.mlClient.callAsCurrentUser`

Reply:
Good call- much better. Added the legacy check in the constructor here: e4c0e64639120c16c78f85258334091d4ec0c6ea